src/module/actor/siegeweapon/data.ts

```diff
@@ -0,0 +1,78 @@
+import { SaveData } from "@actor/creature/data";
+import {
+    ActorSystemData,
+    BaseActorAttributes,
+    BaseActorDataPF2e,
+    BaseActorSourcePF2e,
+    BaseHitPointsData,
+    BaseTraitsData,
+} from "@actor/data/base";
+import { SaveType } from "@actor/types";
+import { ActorSizePF2e } from "@actor/data/size";
+import { SiegeWeaponPF2e } from ".";
+import { SiegeWeaponTrait } from "./types";
+
+/** The stored source data of a siege weapon actor */
+type SiegeWeaponSource = BaseActorSourcePF2e<"siegeweapon", SiegeWeaponSystemData>;
```

Comment:
Separate source from data, even if there's no difference except an alias

Reply:
ok, something more like this?
```
/** The stored source data of a siege weapon actor */
type SiegeWeaponSource = BaseActorSourcePF2e<"siegeWeapon", SiegeWeaponSystemSource>;

interface SiegeWeaponData
    extends Omit<SiegeWeaponSource, "effects" | "flags" | "items" | "prototypeToken">,
        BaseActorDataPF2e<SiegeWeaponPF2e, "siegeWeapon", SiegeWeaponSystemData, SiegeWeaponSource> {}

/** The system-level data of siege weapon actors. */
interface SiegeWeaponSystemSource extends ActorSystemSource {
    attributes: SiegeWeaponAttributesSource;
    details: SiegeWeaponDetailsSource;
    saves: SiegeWeaponSavesSource;
    traits: SiegeWeaponTraitsSource;
}

interface SiegeWeaponSystemData extends SiegeWeaponSystemSource {
    attributes: SiegeWeaponAttributesData;
    details: SiegeWeaponDetailsData;
}
```

Reply:
Yeah, that's it.